parts/windows/windowscalicofunc.ps1

```diff
@@ -62,11 +62,26 @@ function Start-InstallCalico {
     SetConfigParameters -RootDir $CalicoDir -OldString "CALICO_NETWORKING_BACKEND=`"vxlan`"" -NewString "CALICO_NETWORKING_BACKEND=`"none`""
     SetConfigParameters -RootDir $CalicoDir -OldString "KUBE_NETWORK = `"Calico.*`"" -NewString "KUBE_NETWORK = `"azure.*`""
 
-    GetCalicoKubeConfig -RootDir $CalicoDir -CalicoNamespace "kube-system" -SecretName "calico-windows"
+    $calicoNs = GetCalicoNamespace
+    GetCalicoKubeConfig -RootDir $CalicoDir -CalicoNamespace $calicoNs -SecretName "calico-windows"
 
     Write-Log "Install Calico"
 
     pushd $CalicoDir
     .\install-calico.ps1
     popd
+}
+
+function GetCalicoNamespace() {
+    param(
+      [parameter(Mandatory=$false)] $KubeConfigPath = "c:\\k\\config"
+    )
+
+    $name=c:\k\kubectl.exe --kubeconfig=$KubeConfigPath get ns calico-system

```

Comment:
Just wondering with AKS windows nodes, does `c:\k\config` and `c:\k\kubectl.exe` always exist?

Reply:
Yes for now.

Reply:
Did you hit such an issue?
`Error from server (Forbidden): namespaces "calico-system" is forbidden: User "nodeclient" cannot get resource "namespaces" in API group "" in the namespace "calico-system"`

Reply:
What will happen if the command returns an error?

Reply:
@mainred, if your cluster does not have the namespace `calico-system`, you will hit this issue. `name` will be empty and "kube-system" will be returned.
In Windows provisioning scripts, any thrown error will be caught by `catch` in `kuberneteswindowssetup.ps1`. There is no difference here.

Reply:
I have some misunderstanding for your error message. @mainred yes. it is forbidden so we cannot get namespace via kubectl.exe inside Windows nodes.